packages/sitecore-jss/src/layout-service.test.ts

```diff
@@ -12,279 +13,401 @@ import { LayoutServiceData, PlaceholderData } from './dataModels';
-        expect(placeholderData).to.deep.equal({
-          name: 'x1',
-          path: 'x1/x2',
-          elements: [],
+      const res = {
+        setHeader: setHeaderSpy,
+      } as ServerResponse;
+
+      const service = new RestLayoutService({
+        apiHost: 'http://sctest',
+        apiKey: '0FBFF61E-267A-43E3-9252-B77E71CEE4BA',
+        siteName: 'supersite',
+        tracking: false,
+      });
+
+      return service
+        .fetchPlaceholderData('superPh', '/xxx', 'da-DK', req, res)
+        .then((placeholderData: PlaceholderData) => {
+          expect(placeholderData).to.deep.equal({
+            name: 'x1',
+            path: 'x1/x2',
+            elements: [],
+          });
+          expect(setHeaderSpy).to.be.called.with('set-cookie', 'test-set-cookie-value');
         });
-        expect(setHeaderSpy).to.be.called.with('set-cookie', 'test-set-cookie-value');
+    });
+
+    it('should fetch placeholder data using custom fetcher resolver', () => {
+      const fetcherSpy = spy((url: string) => {
+        return new AxiosDataFetcher().fetch<any>(url);
+      });
+
+      mock.onGet().reply(() => {
+        return [
+          200,
+          {
+            name: 'x1',
+            path: 'x1/x2',
+            elements: [],
+          },
+        ];
       });
+
+      const service = new RestLayoutService({
+        apiHost: 'http://sctest',
+        apiKey: '0FBFF61E-267A-43E3-9252-B77E71CEE4BA',
+        siteName: 'supersite',
+        dataFetcherResolver: () => fetcherSpy,
+      });
+
+      return service
+        .fetchPlaceholderData('superPh', '/xxx', 'da-DK')
+        .then((placeholderData: PlaceholderData) => {
+          expect(placeholderData).to.deep.equal({
+            name: 'x1',
+            path: 'x1/x2',
+            elements: [],
+          });
+
+          expect(fetcherSpy).to.be.called.once;
+          expect(fetcherSpy).to.be.called.with(
+            'http://sctest/sitecore/api/layout/placeholder/jss?placeholderName=superPh&item=%2Fxxx&sc_apikey=0FBFF61E-267A-43E3-9252-B77E71CEE4BA&sc_site=supersite&sc_lang=da-DK&tracking=true'
+          );
+        });
+    });
   });
 
-  it('should fetch placeholder data using custom fetcher resolver', () => {
-    const fetcherSpy = spy((url: string) => {
-      return new AxiosDataFetcher().fetch<any>(url);
+  describe('GraphQLLayoutService', () => {
+    beforeEach(() => {
+      nock.cleanAll();
     });
 
-    mock.onGet().reply(() => {
-      return [
-        200,
-        {
-          name: 'x1',
-          path: 'x1/x2',
-          elements: [],
+    it('should fetch layout data', async () => {
+      nock('http://sctest')
+        .post('/graphql', (body) => {
+          return (
+            body.query.replace(/\n|\s/g, '') ===
+            'query{layout(site:"supersite",routePath:"/styleguide",language:"da-DK"){rendered}}'
+          );
+        })
+        .reply(200, {
+          data: {
+            layout: {
```

Comment:
Does this include the latest round of changes to the schema? I thought the layout query will now return an item, and layoutJson is a child of the Item type.

Reply:
@ambrauer But for now we can't use latest schema, Anton Ovcharenko said that this schema is only one which we can test now

Reply:
I wanted to use latest, but Anton said that it's not working now (may be pushed to master branch). Yeah, had another call with him, so decided to add latest schema. I added.